### PR TITLE
[text analytics] generate v3.1-preview.1 code

### LINKED ISSUE #12474: [text analytics] generate v3.1-preview.1 code

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #12474 

Right now I had to hand write multi api code to handle the wiring between different API versions. This is because current python autorest multiapi client generation does not support mixin operations with different signatures across API versions (the service has added parameter `opinion_mining` to the sentiment endpoint)

I have a design to allow different method signatures on the autorest side and will be fixing it there, just don't want to block this on that

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_multiapi.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# ------------------------------------
+from enum import Enum
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Union
+
+
+class ApiVersion(str, Enum):
+    """Key Vault API versions supported by this package"""
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_multiapi.py:13]:
```suggestion
    """Text Analytics API versions supported by this package"""
```

Comment (iscai-msft) [reply sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_multiapi.py:13]: Oh no. Thanks for finding that, and I guess you know where I copied all the code from

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_text_analytics.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from azure.core.credentials import AzureKeyCredential
 from azure.ai.textanalytics import TextAnalyticsClient
 from azure.ai.textanalytics import _models
-from azure.ai.textanalytics._generated import models as _generated_models
+from azure.ai.textanalytics._generated.v3_1_preview_1 import models as _generated_models
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_text_analytics.py:10]: just for my own understanding, we only test 3.1-preview.1?

Comment (iscai-msft) [reply sdk/textanalytics/azure-ai-textanalytics/tests/test_text_analytics.py:10]: for the models repr yes. For all other tests, we mainly test v3.0, but right now I check to see if I can initialize a 3.1-preivew.1 client and that api version is passed correctly to the service https://github.com/Azure/azure-sdk-for-python/blob/8898318b199a09f34645233006b03a3b5446fa00/sdk/textanalytics/azure-ai-textanalytics/tests/test_multiapi.py

**Pull request merged**